eng/common/pipelines/templates/steps/cache-ps-modules.yml

```diff
@@ -0,0 +1,10 @@
+steps:
+- pwsh: |
+    . ./eng/common/scripts/Helpers/PSModule-Helpers.ps1
+    Write-Host "##vso[task.setvariable variable=CachedPSModulePath]$global:CurrentUserModulePath"
+  displayName: Set PS Modules Cache Directory
+- task: Cache@2
+  inputs:
+    key: 'PSModulePath | "$(Agent.OS)" | $(Build.SourcesDirectory)/eng/common/scripts/Import-AzModules.ps1'
```

Comment:
You may like to put in a ```$(CacheSalt)``` cache-busting variable. The idea is that if something happens and you want to build the same commit sha, you can bust the cache if it is blocking you for some reason, but still build the same sha. You don't even need to define the variable on the pipeline yet.

Reply:
I'm curious why the docs don't mention anything about having a safety hatch like this. I know you worked on this feature early on before coming to our team do you think this is a general escape hatch we should include in all our caching usages?

Reply:
The original intention for the feature was to have the ability to queue a build without caching. When queued in this way the build would invalidate the cache. Unfortunately priorities meant that the feature was never implemented on the Pipelines side. I think some of the early design docs mentioned cache salting although obviously not all the internal deliberations are public:

https://github.com/microsoft/azure-pipelines-yaml/blob/master/design/pipeline-caching.md

As you can tell the eventual product evolved from there quite a bit.

Reply:
I'm convinced that having a Salt to invalidate the build cache if needed is a good idea. I will add it.

Reply:
I also like having the CacheSalt option. Similarly, this has me thinking about [cache expiry](https://github.com/microsoft/azure-pipelines-yaml/blob/master/design/pipeline-caching.md#cache-expiry) and the impact that could have on our pipelines. Seems like expiry right now is kind of obtuse. Does the cache task currently show whether it's pulling from cache or doing an update?

Reply:
I don't know about expiry but the task does tell you if it pulled from the cache or if there was a cache miss.